implementations/rust/ockam/ockam_identity/src/channel/decryptor.rs

```diff
@@ -425,46 +446,6 @@ impl<I: IdentityTrait> SecureChannelWorker<I> {
         }
     }
```

Comment:
Would returning a `Result<&mut State<I>>` work here instead?

Reply:
Oh wow github what the hell :raised_eyebrow: 

The comment is about using `take()` to return the State and having to put it back after using. Feels like `&mut State<I>` could make this harder to misuse

Reply:
I wish it could work, but that leads to multiple mut references. So long term solution could be:
1. Reorganize what's included in state to avoid multiple mut references
2. Clone state before working on it
3. Handle errors carefully to always put new state after taking old one
Either way it's out of scope of this PR

Reply:
OK, thanks for the clarification!